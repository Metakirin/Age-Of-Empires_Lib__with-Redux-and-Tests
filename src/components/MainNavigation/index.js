import { paths, titles } from '../../config'
import { NavLink } from 'react-router-dom'
import MainNavLink from '../MainNavlink'

const MainNavigation = () => {
  return (
    <>
      <div className="menu_link">
        <NavLink activeClassName="nav_active" to={paths.home}>
          {titles.home}
        </NavLink>
      </div>
      <MainNavLink path={paths.civilizations} title={titles.civilizations} />
      <MainNavLink path={paths.units} title={titles.units} />
      <MainNavLink path={paths.structures} title={titles.structures} />
      <MainNavLink path={paths.technologies} title={titles.technologies} />
    </>
  )
}
export default MainNavigation
