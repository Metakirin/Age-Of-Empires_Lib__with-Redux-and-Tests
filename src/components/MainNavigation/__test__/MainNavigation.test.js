import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import MainNavigation from '..';
import MainNavLink from '../../MainNavlink'

configure({ adapter: new Adapter() });

describe('Should render correctly "MainNavigation"', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<MainNavigation />);
    });
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1);
    });
    it('component have children div with "menu_link" class', () => {
        expect(wrapper.find('.menu_link')).toHaveLength(1);
    });
    it('component have children component "MainNavLink"', () => {
        expect(wrapper.find(MainNavLink)).toHaveLength(4);
    });
});