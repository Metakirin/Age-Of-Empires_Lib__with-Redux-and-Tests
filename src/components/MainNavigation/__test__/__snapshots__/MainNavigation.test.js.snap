// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly "MainNavigation" to Match Snapshot 1`] = `
<Fragment>
  <div
    className="menu_link"
  >
    <NavLink
      activeClassName="nav_active"
      exact={true}
      to="/"
    >
      Home
    </NavLink>
  </div>
  <Connect(MainNavLink)
    path="/civilizations"
    title="Civilizations"
  />
  <Connect(MainNavLink)
    path="/units"
    title="Units"
  />
  <Connect(MainNavLink)
    path="/structures"
    title="Structures"
  />
  <Connect(MainNavLink)
    path="/technologies"
    title="Technologies"
  />
</Fragment>
`;
